### PR TITLE
[documentation] make it clear that Android target is needed for tests.

### PR DESCRIPTION
**Pull request by atsushieno**

### CODE REVIEW

**Comment by jonpryor** [comment]
The somewhat confusing thing is that while, yes, an Android emulator or device is required, `make run-all-tests` will attempt to *create and launch* an Android emulator if no device is attached:

https://github.com/xamarin/xamarin-android/blob/7c8b6e4ce93b2395f71ff7762afe7aadc416d3f3/build-tools/scripts/TestApks.targets#L21-L50

> you need a running Android emulator or a connected Android device

This implies that the the Android target must already exist before running `make run-all-tests`, which shouldn't be the case.

In short, I'm not sure if this is a net improvement in clarity.

**Comment by atsushieno** [comment]
> make run-all-tests will attempt to create and launch an Android emulator if no device is attached

which never happened. It seems bogus then.

**Comment by atsushieno** [comment]
A funny thing happened: `make run-apk-tests` never launches such an emulator but `msbuild /p:Configuration=Debug  /v:diag /binaryLogger:"/sources/xamarin-android//bin/BuildDebug/msbuild-`date +%s`-run-apk-tests.binlog" build-tools/scripts/RunTests.targets /t:RunApkTests` does (unless /v change matters).

Maybe some escaping issue in the shell?

**Comment by atsushieno** [comment]
Execution observation was not very strict - BOTH resulted in failure of emulator launches, which was exactly caused by this problem https://github.com/xamarin/xamarin-android/pull/1858 . The script seems to automatically create an emulator that is not compatible with platform-tools depending on the setup.

I have rewritten the part, still not to surprise contributors by installing random-looking apps or creating random emulator.
